webapp/app/js/perf.js

```diff
@@ -179,6 +179,83 @@ perf.controller('PerfCtrl', [ '$state', '$stateParams', '$scope', '$rootScope',
       });
     }
 
+    function plotoverviewGraph() {
+      $scope.seriesList.forEach(function(series) {
+        series.flotSeries.points.show = false;
+        series.flotSeries.lines.show = false;
+        if (series.visible){
+          series.flotSeries.lines.show = true;
+        }
+      });
+
+      $scope.overviewPlot = $.plot($("#overview-plot"),
+                              $scope.seriesList.map(
+                                function(series) {
+                                 return series.flotSeries }),
+                                 {
+                                   xaxis: { mode: 'time' },
+                                   selection: { mode: 'x', color: '#97c6e5' },
+                                   series: { shadowSize: 0 },
+                                   lines: { show: true },
+                                   points: { show: false },
+                                   legend: { show: false },
+                                   grid: {
+                                     color: '#cdd6df',
+                                     borderWidth: 2,
+                                     backgroundColor: '#fff',
+                                     hoverable: true,
+                                     clickable: true,
+                                     autoHighlight: false
+                                   }
+                                 });
+
+      $scope.seriesList.forEach(function(series) {
+        series.flotSeries.points.show = series.visible;
+        series.flotSeries.lines.show = false;
+      });
+
+      $("#overview-plot").bind("plotselected", function (event, ranges) {
+        $.each($scope.plot.getXAxes(), function(_, axis) {
+          var opts = axis.options;
+          opts.min = ranges.xaxis.from;
+          opts.max = ranges.xaxis.to;
+        });
+
+        $scope.zoomLevel = [ranges.xaxis.from, ranges.xaxis.to];
+        $scope.plot.setupGrid();
+        $scope.plot.draw();
+        updateURL()
+      });
+    }
+
+    function zoomGraph() {
+      if ($scope.zoomLevel) {
+        var flag = 0;
+        $scope.seriesList.forEach(function(series) {
+          if (series.visible) {
+            flag = 1;
+          }
+        });
+        if (flag == 1) {
```

Comment:
You could replace all of this with:

if (_.find($scope.seriesList, function(series) { return series.visible; })) {
  ...
}

http://underscorejs.org/#find